File: src/041/p041.py
import itertools
from src.utils.py.prime import generate_primes, is_prime


def solve() -> int:
    primes = generate_primes(2800)
    pandigitals = [int("".join(s)) for s in itertools.permutations("7654321")]
    return next(n for n in sorted(pandigitals, reverse=True) if is_prime(n, primes))

File: src/041/p041.go
package problem041

import (
	"strconv"

	"github.com/sunilbpandey/project-euler/src/utils/go/math/primeutils"
	"github.com/sunilbpandey/project-euler/src/utils/go/sliceutils"
	"github.com/sunilbpandey/project-euler/src/utils/go/strutils"
)

func Solve() string {
	primes := primeutils.GeneratePrimes(2800)
	digits := []int{1, 2, 3, 4, 5, 6, 7}
	max := 0
	for perm := range sliceutils.Permutations(digits, len(digits)) {
		n := strutils.Atoi(strutils.JoinInts(perm))
		if primeutils.IsPrime(n, primes) && n > max {
			max = n
		}
	}
	return strconv.Itoa(max)
}
